src/engine/Operation.cpp

```diff
@@ -177,9 +177,7 @@ shared_ptr<const ResultTable> Operation::getResult(bool isRoot,
                << resultNumCols << std::endl;
     return result._resultPointer->resultTable();
   } catch (ad_utility::CancellationException& e) {
-    if (e.operation_.empty()) {
-      e.operation_ = getDescriptor();
-    }
+    e.setOperation(getDescriptor());
```

Comment:
I think now you get the wrong information, as the parent operations will overwrite this.
Either you handle the "operation has already been set, not overwrititing it" issue withing the `CancellationException` class, or you simply rethro as an `AbortException` with the correct `what` as all the other exceptions do.

Reply:
(The old version didn't have the issue as there was the `if operation still empty()` check around it.

Reply:
I see, you do this within the CancellationException. So then this comment is obsolete, please close it.